exercises/practice/leap/.approaches/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+Every fourth year is a leap year (with some exceptions), but let's consider this one condition first. 
+
+To solve the Leap problem, we must determine if a year is evenly divisible by a number or if a reminder of an integer division is zero. 
+Such operation in computing is called [modulo][modulo]. 
+
+Unlike many languages, Elixir does not have [operators][operators] for either integer division or modulo. 
+Instead, it provides [`rem/2`][rem] guard and the [`Integer.mod/2`][mod] function.
+
+The two functions differ in how they work with negative numbers, but since, in this exercise,
+all the numbers are non-negative, both could work, depending on the approach.
+
+## General Solution
+
+To check if a year is divisible by `n`, we can do `rem(year, n) == 0`. We can define a function to make the intent clearer. 
+
+```elixir
+defp divides?(number, divisor), do: rem(number, divisor) == 0 
+```
```

Comment:
I was inspired by my solution to the problem I submitted in the Clojure track. I agree that many of the example or community solutions don't represent anything close to ideal. Sometimes, I see things starred the most, which I would consider an anti-pattern rather than a solution worth showing. 

I think that the approaches are a perfect place where the solutions can be curated to things worth showing rather than cataloguing what's possible. 

As to the divdes? The gain is not big for experienced developers, but for those less experienced, I think it might help with showing intent. When it comes to naming things I'm really inspired by [Kevlin Henny's Seven Ineffective Coding Habits of Many Programmers talk](https://www.youtube.com/watch?v=SUIUZ09mnwM). For me, the code is primarily an intent communication tool. 

But I'll think a bit more about it as I write out the details of the approaches.